tests/utils.py

```diff
@@ -24,6 +24,29 @@ def fig2array(fig):
     return data
 
 
-def match_figure_image(fig, arr_path):
-    """Check whether the two image arrays match."""
+def match_figure_image(fig, arr_path: str):
+    """Check whether the two image arrays match.
+
+    Args:
+        fig (:class:`matplotlib.figure.Figure`): Figure to compare.
+        arr_path (str): Path to the ``numpy`` array file containing the reference image.
+
+    Returns:
+        bool: ``True`` if the images match, ``False`` otherwise.
+    """
     return np.all(fig2array(fig) == np.load(arr_path))
+
+
+def match_figure_close_image(fig, arr_path: str, rtol: float = 0, atol: float = 255):
+    """Check whether the two image arrays match within a tolerance.
+
+    Args:
+        fig (:class:`matplotlib.figure.Figure`): Figure to compare.
+        arr_path (str): Path to the ``numpy`` array file containing the reference image.
+        rtol (float, optional): Relative tolerance for comparison.
+        atol (float, optional): Absolute tolerance for comparison.
+
+    Returns:
+        bool: ``True`` if the images match within the specified tolerances, ``False`` otherwise.
+    """
+    return np.allclose(fig2array(fig), np.load(arr_path), rtol=rtol, atol=atol)
```

Comment:
Instead of saving pictures as arrays, and maintain your own utilities to compare, you could actually save them as PNG images, and just use Matplotlib ones.

https://matplotlib.org/stable/api/testing_api.html#module-matplotlib.testing.compare

Reply:
Yes, since it's querying the file system, you certainly need a temporary path to dump it.

However, that should be fairly doable with [`tempfile`](https://docs.python.org/3/library/tempfile.html) (true, it will load it again - but I'd be willing to pay the performance price to save some maintenance time).

Reply:
Finally I implemented your suggestion @alecandido, but I think that I will not change the `match_figure_image` function at the moment for the circuit drawer because I think it is out of the scope of this ticket and I would like to change it on another dedicated pull request